### ISSUE TITLE
Implement LoopedStringConcatenationToStringBuilder analyzer

### ISSUE THREAD

**Issue by Vannevelj** [Vannevelj/VSDiagnostics#28]
Concatenating strings in a loop causes unnecessary intermediate strings to be created. Instead, use a `StringBuilder` and append each string when calling `.ToString()` which skips the intermediate ones.

This turns this:

	string result = "";
	for(int i = 0; i < 10; i++)
	{
		result += "a";
	}
	Console.WriteLine (result);

into this:

	StringBuilder sb = new StringBuilder();
	for(int i = 0; i < 10; i++)
	{
		sb.Append("a");
	}
	Console.WriteLine (result.ToString());

**Comment by AdrianLThomas** [comment]
Do we only want to see the warning within a loop? i.e. you wouldn't receive the warning for the following:

string exampleString = "Hello " + "Vannevlj " + "!"

**Comment by Vannevelj** [comment]
Exactly, only in loops (`while`, `do-while`, `for`). The scenario you show is exactly the scenario where concatenation is most appropriate.
